pandas/io/parsers.py

```diff
@@ -227,6 +227,21 @@
     Note that the entire file is read into a single DataFrame regardless,
     use the `chunksize` or `iterator` parameter to return the data in chunks.
     (Only valid with C parser)
+compact_ints : boolean, default False
+    DEPRECATED: this argument will be removed in a future version
+
+    If compact_ints is True, then for any column that is of integer dtype,
+    the parser will attempt to cast it as the smallest integer dtype possible,
+    either signed or unsigned depending on the specification from the
+    `use_unsigned` parameter. Selecting this option can help save memory if
+    there are a lot of relatively small integers in the resulting DataFrame.
```

Comment:
But not in consumed memory (which is eg a problem if you are reading in a very large csv file that would need too much memory to read in, although the result may fit in memory), and therefore I find the current wording a bit misleading.

But also see my comment below on the docstring before we further discuss this.